pkg/sources/github/graphql.go

```diff
@@ -0,0 +1,491 @@
+
+				if err := s.chunkComments(ctx, repoInfo, commentsQuery.GetIssueComments(), reporter, cutoffTime); err != nil {
+					return err
+				}
+
+				// update page info for loop
+				issue.Comments.PageInfo = commentsQuery.Repository.Issue.Comments.PageInfo
+			}
+		}
+
+		// paginate issues
+		if !query.Repository.Issues.PageInfo.HasNextPage {
+			ctx.Logger().V(4).Info("Scanned all repository issues with comments", "total_issues_scanned", totalIssues)
+			break
+		}
+
+		// update issues pagination to go to next page
+		vars[issuesPagination] = githubv4.NewString(query.Repository.Issues.PageInfo.EndCursor)
+	}
+
+	return nil
+}
+
+// processPRWithComments process github repo pull requests with inline comments
+func (s *Source) processPRWithComments(ctx context.Context, repoInfo repoInfo, reporter sources.ChunkReporter, cutoffTime *time.Time) error {
+	vars := map[string]any{
+		owner:                 githubv4.String(repoInfo.owner),
+		repository:            githubv4.String(repoInfo.name),
+		pullRequestPerPage:    githubv4.Int(defaultPagination),
+		pullRequestPagination: (*githubv4.String)(nil),
+		commentsPerPage:       githubv4.Int(defaultPagination),
+		commentsPagination:    (*githubv4.String)(nil),
+	}
+
+	var totalPRs int
+
+	// continue loop as long as there are pull requests remaining
+	for {
+		var query pullRequestWithComments
+		err := s.connector.GraphQLClient().Query(ctx, &query, vars)
+		if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &query.RateLimit, err) {
+			continue
+		}
+
+		if err != nil {
+			return fmt.Errorf("error fetching pull requests with comments: %w", err)
+		}
+
+		totalPRs += len(query.GetPullRequests())
+
+		ctx.Logger().V(5).Info("Scanning pull requests",
+			"total_pull_requests", len(query.GetPullRequests()))
+
+		if err := s.chunkGraphqlPullRequests(ctx, repoInfo, query.GetPullRequests(), reporter); err != nil {
+			return err
+		}
+
+		// process each pr comments
+		for _, pr := range query.GetPullRequests() {
+			ctx.Logger().V(5).Info("Scanning pull request comments",
+				"pull_request_no", pr.Number,
+				"total_comments", len(pr.Comments.Nodes))
+
+			if err := s.chunkComments(ctx, repoInfo, pr.GetPRComments(), reporter, cutoffTime); err != nil {
+				return err
+			}
+
+			// if a pull request has more than 100 comments - some interns pull request might endup here :)
+			for pr.Comments.PageInfo.HasNextPage {
+				// request this pull request more comments
+				var commentQuery singlePRComments
+				singlePRVars := map[string]any{
+					owner:              githubv4.String(repoInfo.owner),
+					repository:         githubv4.String(repoInfo.name),
+					pullRequestNumber:  pr.Number,
+					commentsPerPage:    githubv4.Int(defaultPagination),
+					commentsPagination: pr.Comments.PageInfo.EndCursor,
+				}
+
+				err := s.connector.GraphQLClient().Query(ctx, &commentQuery, singlePRVars)
+				if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &query.RateLimit, err) {
```

Comment:
```suggestion
				if s.handleGraphqlRateLimitWithChunkReporter(ctx, reporter, &commentQuery.RateLimit, err) {
```

Similar to another comment I made.   It's not obvious to me from looking here which rate limit we should be respecting.  Some inline comments might help.

If we _aren't_ interested in respecting separate rate limits for each query object separately, I don't understand why the rate limit object is attached to the query object.  (maybe out of scope for this?).